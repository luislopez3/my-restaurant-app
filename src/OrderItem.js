import React from "react";

export default function OrderItem({ index, item = {}, loadItems }) {
  function updateItem(quantity, item_id) {
    fetch(`http://localhost:5000/order_items/${item_id}`, {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ quantity }),
    })
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        loadItems();
      });
  }

  function removeItem(item) {
    fetch(`http://localhost:5000/order_items/${item.id}`, {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json",
      },
    }).then((data) => {
      loadItems();
    });
  }

  return (
    <div className="my-order">
      <ul key={index}>
        <h2>{item.type} Menu</h2>
        <h3>Item: {item.name}</h3>
        <img className="menu-img" src={item.image_url} alt={item.name}></img>
        <p>
          Quantity:{" "}
          <input
            onChange={(e) => updateItem(e.target.value, item.id)}
            type="number"
            value={item.quantity}
          />
        </p>

        <p>
          <button onClick={() => removeItem(item)}>Remove Item</button>
        </p>
        <p>Price: ${item.price}</p>
      </ul>
    </div>
  );
}
