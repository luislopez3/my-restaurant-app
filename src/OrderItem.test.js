import React from "react";
import { create } from "react-test-renderer";
import OrderItem from "./OrderItem";
import { render } from "@testing-library/react";

describe("OrderItem component", () => {
  test("renders OrderItem w/o an error", () => {
    render(<OrderItem />);
  });

  test("Matches the snapshot", () => {
    const wrapper = create(<OrderItem />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});