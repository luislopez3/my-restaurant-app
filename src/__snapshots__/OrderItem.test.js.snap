// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem component Matches the snapshot 1`] = `
<ul>
  <p>
     Menu
  </p>
  <p>
    Menu Item: 
  </p>
  <img />
  <p>
    Quantity:
     
    <input
      onChange={[Function]}
      type="number"
    />
  </p>
  <p>
    <button
      onClick={[Function]}
    >
      Remove Item
    </button>
  </p>
  <p>
    Price: $
  </p>
</ul>
`;
